src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -160,36 +171,23 @@ export class AccessibleBufferWidget extends DisposableStore {
 		this._xtermElement.classList.remove(CssClass.Hide);
 	}
 
-	private async _updateModel(insertion?: boolean): Promise<ITextModel> {
-		let model = this._editorWidget.getModel();
-		const lineCount = model?.getLineCount() ?? 0;
-		if (insertion && model && lineCount > this._xterm.raw.rows) {
-			const lineNumber = lineCount + 1;
-			model.pushEditOperations(null, [{
-				range: { startLineNumber: lineNumber, endLineNumber: lineNumber, startColumn: 1, endColumn: 1 }, text: this._getContent(1)
-			}], () => []);
-		} else {
-			model = await this._getTextModel(this._instance.resource.with({ fragment: this._getContent() }));
-		}
-		if (!model) {
-			throw new Error('Could not create accessible buffer editor model');
+	private async _updateEditor(): Promise<void> {
+		if (this._inProgressUpdate) {
+			return;
 		}
-		this._editorWidget.setModel(model);
-		return model;
-	}
-
-	private async _updateEditor(insertion?: boolean): Promise<void> {
-		const model = await this._updateModel(insertion);
+		this._inProgressUpdate = true;
```

Comment:
This could lead to a race condition, you if an update request occurs when `_inProgressUpdate` it true, you will want to queue up another `_updateEditor` instead of ignoring it.

Reply:
I don't think so as there's duplication that occurs for `onWriteParsed`

Reply:
oh actually that's not needed anymore

Reply:
I don't think this is resolved yet? What happens when `_updateEditor` is called and then it's called again before `_updateModel` is resolved?